rclcpp/include/rclcpp/any_service_callback.hpp

```diff
@@ -100,6 +100,7 @@ class AnyServiceCallback
     TRACEPOINT(callback_end, (const void *)this);
   }
 
+#ifndef TRACETOOLS_DISABLED
```

Comment:
Is it a good idea to change the API based on the define? Code calling this API must then also ifdef the call. Also building with/without the define changes API/ABI.

How about putting the ifdef only around the content of the function?

Reply:
Done in 25dda366a2ed498a892d7fc13b2c055c5bc5433f (while keeping the `#ifndef`s around the function calls)